packages/block-editor/src/components/url-input/index.js

```diff
@@ -22,230 +22,213 @@ import { withInstanceId, withSafeTimeout, compose } from '@wordpress/compose';
-			! this.scrollingIntoView
+			!! selectedSuggestionNode &&
+			! scrollingIntoView.current
 		) {
-			this.scrollingIntoView = true;
-			scrollIntoView(
-				this.suggestionNodes[ selectedSuggestion ],
-				this.autocompleteRef.current,
-				{
-					onlyScrollIfNeeded: true,
-				}
-			);
+			scrollingIntoView.current = true;
+			scrollIntoView( selectedSuggestionNode, autocompleteRef.current, {
+				onlyScrollIfNeeded: true,
+			} );
 
-			this.props.setTimeout( () => {
-				this.scrollingIntoView = false;
+			setTimeout( () => {
+				scrollingIntoView.current = false;
 			}, 100 );
 		}
 
-		// Only attempt an update on suggestions if the input value has actually changed.
-		if (
-			prevProps.value !== value &&
-			this.shouldShowInitialSuggestions()
-		) {
-			this.updateSuggestions();
+		if ( prevValue.current !== value && shouldShowInitialSuggestions() ) {
+			updateSuggestions();
 		}
-	}
+	}, [ showSuggestions, value ] );
 
-	componentDidMount() {
-		if ( this.shouldShowInitialSuggestions() ) {
-			this.updateSuggestions();
+	useEffect( () => {
+		if ( shouldShowInitialSuggestions() ) {
+			updateSuggestions();
 		}
+
+		// cleanup
+		return () => {
+			clearSuggestionsRequest();
+		};
+	}, [] );
+
+	function shouldShowInitialSuggestions() {
+		return (
+			! isUpdatingSuggestions() &&
+			showInitialSuggestions &&
+			isEmpty( value ) &&
+			isEmpty( suggestions )
+		);
 	}
 
-	componentWillUnmount() {
-		delete this.suggestionsRequest;
+	function clearSuggestionsRequest() {
+		pendingRequest.current = null;
 	}
 
-	bindSuggestionNode( index ) {
+	function bindSuggestionNode( index ) {
 		return ( ref ) => {
-			this.suggestionNodes[ index ] = ref;
+			suggestionNodes.current[ index ] = ref;
 		};
 	}
 
-	shouldShowInitialSuggestions() {
-		const { suggestions } = this.state;
-		const {
-			__experimentalShowInitialSuggestions = false,
-			value,
-		} = this.props;
-		return (
-			! this.isUpdatingSuggestions &&
-			__experimentalShowInitialSuggestions &&
-			! ( value && value.length ) &&
-			! ( suggestions && suggestions.length )
-		);
+	function isUpdatingSuggestions() {
+		return pendingRequest.current;
 	}
 
-	updateSuggestions( value = '' ) {
-		const {
-			__experimentalFetchLinkSuggestions: fetchLinkSuggestions,
-			__experimentalHandleURLSuggestions: handleURLSuggestions,
-		} = this.props;
+	function clearUpdateSuggestions() {
+		setLoading( false );
+		clearSuggestionsRequest();
+	}
 
+	function allowSuggestionsRequest( search = '' ) {
 		if ( ! fetchLinkSuggestions ) {
-			return;
+			return false;
 		}
 
-		const isInitialSuggestions = ! ( value && value.length );
-
-		// Allow a suggestions request if:
-		// - there are at least 2 characters in the search input (except manual searches where
-		//   search input length is not required to trigger a fetch)
-		// - this is a direct entry (eg: a URL)
-		if (
-			! isInitialSuggestions &&
-			( value.length < 2 || ( ! handleURLSuggestions && isURL( value ) ) )
```

Comment:
This was tricky to refactor since it seems that the `isURL` check is redundant. If you assume `! handleURLSuggestions` is true the statement becomes `value.length < 2 || isURL(value)`. `isURL` will return false if the string does not include the protocols so `value` must be _at least_ `http://` in which case `value.length` will be greater than 2 and the suggestion request will be made.

I tried to interpret the intended behavior as _if the search value is a valid url and we should not handle url suggestions, don't bother making the suggestion request_ here [#](https://github.com/WordPress/gutenberg/pull/27559/files#diff-6a61cdc612f2906b7754c38eb8d671d60cf4c4a07ecb4a283792a3d450947b7cR141)